src/components/Settings/SettingsRelationships.js

```diff
@@ -0,0 +1,341 @@
+import {Alert, Image, Text, TextInput, View} from "react-native";
+import {viewStyles, textStyles} from "../../styles/settings/settings";
+import React, {useCallback, useEffect} from "react";
+import Pressable from "react-native/Libraries/Components/Pressable/Pressable";
+import {useDebounce} from "use-debounce";
+import inatjs from "inaturalistjs";
+import {Picker} from "@react-native-picker/picker";
+import {colors} from "../../styles/global";
+import CheckBox from "@react-native-community/checkbox";
+import UserSearchInput from "./UserSearchInput";
+
+const FOLLOWING = {
+  all: "All",
+  yes: "Yes",
+  no: "No"
+};
+
+const TRUSTED = {
+  all: "All",
+  yes: "Yes",
+  no: "No"
+};
+
+const SORT_BY = {
+  recently_added: "Recently Added",
+  earliest_added: "Earliest Added",
+  a_to_z: "A to Z",
+  z_to_a: "Z to A"
+};
+
+
+const SettingsRelationships = ( { accessToken, settings, onRefreshUser } ): React.Node => {
+  const [relationshipResults, setRelationshipResults] = React.useState( [] );
+  const [userSearch, setUserSearch] = React.useState( "" );
+  // So we'll start searching only once the user finished typing
+  const [finalUserSearch] = useDebounce( userSearch, 500 );
+  const [following, setFollowing] = React.useState( "all" );
+  const [trusted, setTrusted] = React.useState( "all" );
+  const [sortBy, setSortBy] = React.useState( "desc" );
+  const [page, setPage] = React.useState( 1 );
+  const [totalPages, setTotalPages] = React.useState( 1 );
+  const [blockedUsers, setBlockedUsers] = React.useState( [] );
+  const [mutedUsers, setMutedUsers] = React.useState( [] );
+
+  const refreshRelationships = useCallback( async () => {
+    try {
+      const orderBy = ["a_to_z", "z_to_a"].includes( sortBy ) ? "" : "users.login";
+      const order = ["z_to_a", "recently_added"].includes( sortBy ) ? "desc" : "asc";
+      console.log( {q: finalUserSearch, following, trusted, order_by: orderBy, order: order } );
+      const response = await inatjs.relationships.search(
+        {q: finalUserSearch, following, trusted, order_by: orderBy, order: order, per_page: 10, page },
+        { api_token: accessToken} );
+      console.log( response );
+      setRelationshipResults( response.results );
+      setTotalPages( Math.ceil( response.total_results / response.per_page ) );
+    } catch ( e ) {
+      console.error( JSON.stringify( e ) );
+    }
+  }, [accessToken, finalUserSearch, following, page, sortBy, trusted] );
+
+  useEffect( () => {
+    refreshRelationships();
+  }, [accessToken, finalUserSearch, following, trusted, sortBy, page, refreshRelationships] );
+
+  useEffect( () => {
+    const getBlockedUsers = async () => {
+      try {
+        const responses = await Promise.all( settings.blocked_user_ids.map( ( userId ) => inatjs.users.fetch( userId ) ) );
```

Comment:
I'm getting an `undefined is not an object (evaluating 'settings.blocked_user_ids.length')` error here, so I can't load the relationships tab. I'm guessing the same error will pop up for `muted_user_ids`.

Reply:
v2 API issue - `users/me` is missing muted_user_ids and blocked_user_ids

Reply:
Still not working, but for a different reason: you're querying the relationships with values the API doesn't understand.